cli_tools/gce_ovf_import/daisy_utils/data_disk_workflow_updater.go

```diff
@@ -40,13 +40,11 @@ func AddDiskImportSteps(w *daisy.Workflow, dataDiskInfos []ovfutils.DiskInfo) {
 	for i, dataDiskInfo := range dataDiskInfos {
 		dataDiskIndex := i + 1
 		dataDiskFilePath := dataDiskInfo.FilePath
-		diskNames = append(
-			diskNames,
-			fmt.Sprintf("%v-data-disk-%v", w.Vars["instance_name"].Value, dataDiskIndex))
+		diskNames = append(diskNames, generateDataDiskName(w.Vars["instance_name"].Value, dataDiskIndex))
 
 		setupDataDiskStepName := fmt.Sprintf("setup-data-disk-%v", dataDiskIndex)
-		diskImporterDiskName := fmt.Sprintf("disk-importer-%v", dataDiskIndex)
-		scratchDiskDiskName := fmt.Sprintf("disk-importer-scratch-%v-%v", dataDiskIndex, w.Vars["instance_name"].Value)
+		diskImporterDiskName := fmt.Sprintf("disk-importer-%v-%v", dataDiskIndex, w.ID())
+		scratchDiskDiskName := fmt.Sprintf("disk-importer-scratch-%v-%v", dataDiskIndex, w.ID())
```

Comment:
I'd recommend using generateDataDiskName here as well -- might as well benefit from the safety that it gives to protect yourself from future changes to w.ID().

Reply:
These two disk names will never exceed 63 characters, since dataDiskIndex will be up to 2-3 characters long once converted to string and w.ID() is 5 characters.

Reply:
I'm aware that it's fine now, but my point is that you can make your code defensive against future changes to w.ID().

Reply:
I feel this is an overkill. We control what w.ID() is. 

Even if it gets switched under us it to something really long, like UUID, which is 36 characters when formatted as a string, we still have 39 characters of available space disk-importer-scratch and 47 characters for disk-importer.

Reply:
Also, the tests below are checking that all disk names are valid. So, if someone changes w.ID() to something longer than 39 characters (again, unlikely), data_disk_workflow_updater_test.go and disk naming can be fixed at that time. Since, user doesn't control any part of these two disk's names, I don't think we need to do anything now.

Reply:
I'm encouraging defensive coding, and it would also ensure consistency to the reader (hey, we sanitize all disk names!). Is there a downside that I'm not seeing?

Reply:
Just to make it clear: you're asking me to write code that is defensive against itself?